docs/spec_draft.md

```diff
@@ -341,6 +351,174 @@ only exists to establish data dependencies from `result` to `inputs`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.all_gather
+
+### Semantics
+
+For each replica subgroup in `replica_groups`, the operation
+[concatenates](#stablehloconcatenate) the replicas of `input` tensor formed
+using the subgroup along `all_gather_dim` dimension following the order of the
+replicas in the subgroup and produces a `result` tensor.
+
+`channel_handle`, an optional input, is a handle given to a user to represent a
+communication channel between two computations via a
+[stablehlo.send](#stablehlosend) and [stablehlo.recv](#stablehlorecv)
+instruction pair or via collective instructions (e.g.
```

Comment:
Are you using "instruction" and "operation" interchangeably or is there an implied difference between the two?